mGui/core/__init__.py

```diff
@@ -122,19 +122,27 @@ def __init__(self, key=None, **kwargs):
         # key is our internal name
         self.key = self.widget.split("|")[-1]
 
+        # Used to track if the control was created in a modal context.
+        # Note that unlike Nested.modal, this is a 'private' field
+        self._is_modal = False
+
         # Event objects
         self.callbacks = {}
 
         # add us to the current layout under our own key name
         Layout.add_current(self)
         self.onDeleted += self.forget
 
-    def register_callback(self, callbackName, event):
+        # a weak reference to our parent, will be added when
+        # this widget is added to a control
+        self._parent = None
```

Comment:
Should move this to before `Layout.add_current(self)` as it currently overwrites the parent value while inside the 'with-block'.

But as it stands if I try to hook a command right now I get a MayaEvent not Event, because `Control.parent is None`.